src/lib.rs

```diff
@@ -175,61 +195,74 @@ impl IPTables {
     #[cfg(target_os = "linux")]
     pub fn chain_exists(&self, table: &str, chain: &str) -> IPTResult<bool> {
         match self.run(&["-t", table, "-L", chain]) {
-            Ok(output) => Ok(output.status.success()),
+            Ok(_) => Ok(true),
+            Err(IPTError::BadExitStatus(1)) => Ok(false),
             Err(err) => Err(err),
         }
     }
 
     /// Inserts `rule` in the `position` to the table/chain.
     /// Returns `true` if the rule is inserted.
-    pub fn insert(&self, table: &str, chain: &str, rule: &str, position: i32) -> IPTResult<bool> {
-        match self.run(&[&["-t", table, "-I", chain, &position.to_string()], rule.split_quoted().as_slice()].concat()) {
-            Ok(output) => Ok(output.status.success()),
-            Err(err) => Err(err),
-        }
+    pub fn insert(&self, table: &str, chain: &str, rule: &str, position: i32) -> IPTResult<()> {
```

Comment:
Here's an example where the result type should maybe be `IPResult<bool>` instead. Thoughts?